frontend/src/store/objectStore.ts

```diff
@@ -214,6 +218,20 @@ export const useObjectStore = defineStore('objectStore', () => {
     }
   }
 
+  // Permission guards for the buttons
+  function isActionAllowed(objectPermissions: Permission[], perm: string, userId?: string) {
+    // If you have bucket manage you can do it all
+    // OR if you have the specified permission on the bucket
+    // OR if you have the specified permission on the object
+    return (
+      selectedBucketPermissionsForUser.value.some((bp) => bp.permCode === Permissions.MANAGE && bp.userId === userId)
+      ||
+      selectedBucketPermissionsForUser.value.some((bp) => bp.permCode === perm && bp.userId === userId)
+      ||
+      objectPermissions.some((op) => op.permCode === perm && op.userId === userId)
+    );
+  }
```

Comment:
The way the discretionary access control model was designed in COMS, the `MANAGE` permission only grants the user the ability to manipulate permissions for that specific resource. It does NOT implicitly grant the other four permissions by virtue of existing. All 5 of our CRUDM permissions are meant to be mutually exclusive to each other, and must be explicitly assigned to each individual.